clients/java/src/main/java/io/camunda/zeebe/client/impl/worker/JobWorkerImpl.java

```diff
@@ -194,32 +194,43 @@ private void onPollSuccess(final JobPoller jobPoller, final int activatedJobs) {
     // first release, then lookup remaining jobs, to allow handleJobFinished() to poll
     releaseJobPoller(jobPoller);
```

Comment:
:x: I think we cannot release the poller anymore, or not yet, as it allows `handleJobFinished` to poll again.

Reply:
It needs to be released in the else block (since in the if block, backoff will release it anyway).

Reply:
That means our test is not catching it properly, so we need to also review this and why the test isn't catching it